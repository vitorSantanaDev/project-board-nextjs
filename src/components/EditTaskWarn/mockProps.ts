export const handleClickCancelEdit = jest.fn()
