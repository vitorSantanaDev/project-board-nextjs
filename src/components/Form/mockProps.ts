export const formProps = {
  handleAddTaskSubmit: jest.fn(),
  handleChangeInput: jest.fn(),
  input: ''
}
