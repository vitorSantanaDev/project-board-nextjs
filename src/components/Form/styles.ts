import styled from 'styled-components'

export const Wrapper = styled.form`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const InputElement = styled.input`
  width: 90%;
  height: 50px;

  padding: 8px 16px;

  border-radius: 6px;
  border: 1px solid #424242;

  color: #fff;
  background-color: #20212c;
`

export const ButtonSubmit = styled.button`
  width: 8%;
  height: 50px;
  cursor: pointer;
  border-radius: 6px;

  margin-left: 16px;

  background-color: #ffb800;
  transition: ease-in-out 0.3s;

  &:hover {
    transform: scale(1.05);
  }
`
