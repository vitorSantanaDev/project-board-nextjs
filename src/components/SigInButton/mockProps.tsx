export const mockProps = {
  session: true,
  onClick: jest.fn()
}
