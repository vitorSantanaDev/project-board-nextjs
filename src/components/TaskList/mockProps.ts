export const tasksProps = [
  {
    id: '3896433',
    userId: '8458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '39033',
    userId: '8456458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '38933',
    userId: '865658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '3366553',
    userId: '85654658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '337763',
    userId: '88',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  }
]
