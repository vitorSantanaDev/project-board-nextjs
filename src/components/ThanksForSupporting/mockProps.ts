export const mockProps = {
  lastDonateUser: `Sua última doação foi a ${new Date().toLocaleDateString()}`
}
