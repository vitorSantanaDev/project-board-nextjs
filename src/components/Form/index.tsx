import { FormEvent, SetStateAction } from 'react'
import { FiPlus } from 'react-icons/fi'
import * as S from './styles'

interface FormProps {
  handleAddTaskSubmit: (event: FormEvent) => void
  handleChangeInput: (event: {
    target: { value: SetStateAction<string> }
  }) => void
  input: string
}

const Form: React.FC<FormProps> = ({
  handleAddTaskSubmit,
  handleChangeInput,
  input
}) => {
  return (
    <S.Wrapper onSubmit={handleAddTaskSubmit}>
      <S.InputElement
        type="text"
        value={input}
        onChange={handleChangeInput}
        placeholder="Digite sua tarefa"
      />
      <S.ButtonSubmit data-testid="buttonSubmit" type="submit">
        <FiPlus size={25} color="#17181f" />
      </S.ButtonSubmit>
    </S.Wrapper>
  )
}

export default Form
