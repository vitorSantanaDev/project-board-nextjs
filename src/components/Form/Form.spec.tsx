import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Form from '.'
import { formProps } from './mockProps'

describe('<Form />', () => {
  it('should render the Form', () => {
    render(<Form {...formProps} />)
    const input = screen.getByPlaceholderText('Digite sua tarefa')
    const button = screen.getByTestId('buttonSubmit')

    expect(input).toBeInTheDocument()
    expect(button).toBeInTheDocument()
  })
  it('should must call the handleChangeInput function, when typing some value in the input', () => {
    render(<Form {...formProps} />)
    const input = screen.getByPlaceholderText('Digite sua tarefa')

    userEvent.type(input, 'new Task')
    expect(formProps.handleChangeInput).toBeCalledTimes(8)
  })
  it('should call the handleAddTaskSubmit function and clear the input when you click the button', () => {
    render(<Form {...formProps} />)
    const input = screen.getByPlaceholderText('Digite sua tarefa')
    const button = screen.getByTestId('buttonSubmit')

    userEvent.click(button)
    expect(formProps.handleAddTaskSubmit).toBeCalledTimes(1)
    userEvent.type(input, '')
  })

  it('should match snapshot', () => {
    const { container } = render(<Form {...formProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
