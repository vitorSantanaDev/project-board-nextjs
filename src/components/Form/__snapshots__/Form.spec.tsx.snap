// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 90%;
  height: 50px;
  padding: 8px 16px;
  border-radius: 6px;
  border: 1px solid #424242;
  color: #fff;
  background-color: #20212c;
}

.c2 {
  width: 8%;
  height: 50px;
  cursor: pointer;
  border-radius: 6px;
  margin-left: 16px;
  background-color: #ffb800;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
}

.c2:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

<form
  class="c0"
>
  <input
    class="c1"
    placeholder="Digite sua tarefa"
    type="text"
    value=""
  />
  <button
    class="c2"
    data-testid="buttonSubmit"
    type="submit"
  >
    <svg
      color="#17181f"
      fill="none"
      height="25"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(23, 24, 31);"
      viewBox="0 0 24 24"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="5"
        y2="19"
      />
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
  </button>
</form>
`;
